Add new option `autosaveBans`

nerdalertdk opened the pull request:
Its not an bug, but more nice of us to do.

If you forget $this->writeBans(); after an ban, the bans.txt will not save the ban when the server restarts and you bans.txt will be empty or the player you banned is not there anymore

nerdalertdk committed:
Added writeBans to addBan/banPlayer/removeBan

Its not an bug, but more nice of us to do.

If you forget $this->writeBans(); after an ban, the bans.txt will not save the ban when the server restarts and you bans.txt will be empty or the player you banned is not there anymore

nerdalertdk commented:
We could add "autosave" as an options by using the $Options array.

Nizarii commented:
That's a great idea 👍 Please add this to your PR, it will be part of the next minor release 2.2. There is also a little issue in your PR: `writeBans()` should be called after `reconnect()` (#12) in `banPlayer()`. Thank you for contribution.

nerdalertdk committed:
Added auto save bans as an options

changed auto save bans so its an options, that way it will not break current systems

nerdalertdk committed:
Update README.md

nerdalertdk commented:
Forget checkOptionTypes

Nizarii commented:
I have already added it ;)

Nizarii commented:
I have also changed `autosavebans` to `autosaveBans`, so it's easier to read. What do you think about setting the default value of the new option to `true`? On the one hand, it might come to a double `writeBans()` call for users, who already have their project setup. On the other hand, we will prevent accidents :/

nerdalertdk commented:
I set it to false AS default, so it did not break current installs. 

Maybe highlight it in the readme file?

nerdalertdk commented:
Or since you are doing and major release set it as true and add an comment to the changelog

Nizarii commented:
I guess it's propably the best choice not to break current installations, because I am just doing a minor release. I will highlight the new feature in the Readme file.
